### ISSUE TITLE
tar: Option --wildcards is not supported

### ISSUE THREAD

**Issue by dsampson01** [extremeshok/clamav-unofficial-sigs#303]
Running FreeBSD 12.0-RELEASE.

When executing clamav-unofficial-sigs.sh from cron we see the following error message:

tar: Option --wildcards is not supported

Usage:

  List:    tar -tf <archive-filename>

  Extract: tar -xf <archive-filename>

  Create:  tar -cf <archive-filename> [filenames...]

  Help:    tar --help

It seems the --wildcard parameter isn't supported in FreeBSD 12.0.

~Doug
--
Doug Sampson
dsampson01 at gmail dot com

**Comment by msimerson** [comment]
Same issue here.

**Comment by tiiiecherle** [comment]
Same issue on macOS, which leads to
`Clamscan reports LinuxMalwareDetect rfxn.ndb database integrity tested BAD`

When installing `gtar` via homebrew and changing 
`tar_bin="$(command -v tar 2> /dev/null)"`
to
`tar_bin="$(command -v gtar 2> /dev/null)"`
everything works as expected.

It seems a few versions of tar do not have the `--wildcards` option.

Would be nice if the command that uses this could be reworked to be compatible with more/all tar versions.

Thanks

**Comment by dsampson01** [comment]
The FreeBSD OS comes with tar as installed. However, it doesn't contain the
wildcard parameter. Installing gtar either as a package or port solves this
issue. Thus there are two versions of tar installed. The
clamav-unofficial-sigs.sh (as of version 7.0.1) checks for the existence of
tar and upon finding it on my system proceeds to use it instead of gtar.

I would think one could specify the desired tar version to use as part of
the user config file.

~Doug

**Comment by tiiiecherle** [comment]
As described in [#305](https://github.com/extremeshok/clamav-unofficial-sigs/issues/305) I put together a little install script for macOS and included a workaround by doing:

```
HOMEBREW_PATH=$(brew --prefix)
HOMEBREW_BIN_PATH=""$HOMEBREW_PATH"/bin"

if command -v gtar &> /dev/null
then
	# installed
	sed -i '' 's/command -v tar/command -v gtar/g' ""$HOMEBREW_BIN_PATH"/clamav-unofficial-sigs.sh"
else
	# not installed
	:
fi
```

**Comment by extremeshok** [comment]
fixed: 552a147a114151baa57ea07dda284a91c15d43ca

**Comment by mundschenk-at** [comment]
I am a bit late to the party, but I maintain (sort of) the FreeBSD port of this package. As far as I've been able to tell, `gtar` is not necessary for use on FreeBSD. 

Changing `$tar_bin --strip-components=1 --wildcards --overwrite -xzf "${work_dir_linuxmalwaredetect}/sigpack.tgz" --directory "${work_dir_linuxmalwaredetect}" */rfxn.*` to `$tar_bin --strip-components=1 -xzf "${work_dir_linuxmalwaredetect}/sigpack.tgz" --directory "${work_dir_linuxmalwaredetect}" "*/rfxn.*"` works fine. (Overwriting files is the default mode for FreeBSD `tar` and wildcards are supported without extra flags.)

**Comment by extremeshok** [comment]
fixed : 7.2.4
